cognite/neat/graph/loaders/rdf_to_dms.py

```diff
@@ -0,0 +1,187 @@
+
+        create_nodes()
+
+
+def upload_edges(
+    client: CogniteClient,
+    edges: list[EdgeApply],
+    batch_size: int = 5000,
+    max_retries: int = 1,
+    retry_delay: int = 3,
+):
+    """Uploads edges to CDF
+
+    Args:
+        client: Instance of CogniteClient
+        edges: List of edges to upload to CDF
+        batch_size: Size of batch. Defaults to 5000.
+        max_retries: Maximum times to retry the upload. Defaults to 1.
+        retry_delay: Time delay before retrying the upload. Defaults to 3.
+    """
+    if batch_size:
+        logging.info(f"Uploading edges in batches of {batch_size}")
+        _micro_batch_push(
+            client,
+            edges,
+            batch_size,
+            push_type="edges",
+            message="Upload",
+            max_retries=max_retries,
+            retry_delay=retry_delay,
+        )
+
+    else:
+        logging.info("Batch size not set, pushing all edges to CDF in one go!")
+
+        @retry_decorator(max_retries=max_retries, retry_delay=retry_delay, component_name="create-edges")
+        def create_nodes():
+            client.data_modeling.instances.apply(edges=edges, auto_create_start_nodes=True, auto_create_end_nodes=True)
+
+        create_nodes()
+
+
+def _micro_batch_push(
+    client: CogniteClient,
+    nodes_or_edges: list[NodeApply] | list[EdgeApply],
+    batch_size: int = 1000,
+    push_type: str = "nodes",
+    message: str = "Upload",
+    max_retries: int = 1,
+    retry_delay: int = 3,
+):
+    """Uploads nodes or edges in batches
+
+    Args:
+        client: Instance of CogniteClient
+        nodes_or_edges: List of nodes or edges
+        batch_size: Size of batch. Defaults to 1000.
+        push_type: Type of push, either "nodes" or "edges". Defaults to "nodes".
+        message: Message to logged. Defaults to "Upload".
+        max_retries: Maximum times to retry the upload. Defaults to 1.
+        retry_delay: Time delay before retrying the upload. Defaults to 3.
+    """
+    total = len(nodes_or_edges)
+    counter = 0
+
+    if push_type not in ["nodes", "edges"]:
+        logging.info(f"push_type {push_type} not supported")
+        raise ValueError(f"push_type {push_type} not supported")
+
+    for batch in chunker(nodes_or_edges, batch_size):
+        counter += len(batch)
+        start_time = datetime_utc_now()
+
+        @retry_decorator(max_retries=max_retries, retry_delay=retry_delay, component_name=f"microbatch-{push_type}")
+        def upsert_nodes_or_edges(batch):
+            if push_type == "nodes":
+                client.data_modeling.instances.apply(
+                    nodes=nodes_or_edges, auto_create_start_nodes=True, auto_create_end_nodes=True
+                )
+            elif push_type == "edges":
+                client.data_modeling.instances.apply(
+                    edges=nodes_or_edges, auto_create_start_nodes=True, auto_create_end_nodes=True
+                )
```

Comment:
1. All nodes/edges are  uploaded every time. Fixed by using correct variable
2.  In addition, suggest renaming batch to avoid shadowing outer name. 
3.  Remove `auto_create_start_nodes=True, auto_create_end_nodes=True` for node as they have no effect. 

```suggestion
        def upsert_nodes_or_edges(upload_batch):
            if push_type == "nodes":
                client.data_modeling.instances.apply(nodes=upload_batch)
            elif push_type == "edges":
                client.data_modeling.instances.apply(
                    edges=upload_batch, auto_create_start_nodes=True, auto_create_end_nodes=True
                )
```